runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -730,6 +730,9 @@ function M.get_virtual_text_chunks_for_line(bufnr, line, line_diags, opts)
   opts = opts or {}
   local prefix = opts.prefix or "■"
   local spacing = opts.spacing or 4
+  local show_server_name = opts.show_server_name or false
+  local server_name_left = opts.server_name_left or '«'
+  local server_name_right = opts.server_name_left or '»'
```

Comment:
Instead of doing this, we should do something where we allow people to pass either a format string (`"<<%s>>"`) or a function that takes some arguments to generate this string.